packages/webapp/src/containers/Task/saga.js

```diff
@@ -616,6 +616,12 @@ const getCompleteIrrigationTaskBody = (task_translation_key) => (data) => {
         ? data.taskData[taskType]?.irrigation_task_type_other
         : data.taskData[taskType]?.irrigation_type_name;
       data.taskData[taskType].location_id = data.location_id;
+      if (data.taskData[taskType]?.is_modified_on_completion) {
+        if (data.taskData[taskType]?.default_irrigation_task_type_location)
+          data.taskData[taskType].default_irrigation_task_type_location = false;
+        if (data.taskData[taskType]?.default_irrigation_task_type_measurement)
+          data.taskData[taskType].default_irrigation_task_type_measurement = false;
```

Comment:
@Duncan-Brain I think this was what Craig did in his last PR, but do we need to update these when completing a task?

Reply:
@SayakaOno It is similar and continues the work(his file changes: https://github.com/LiteFarmOrg/LiteFarm/pull/2473/files). 

These fields show the user what they did on that task during task creation. It should not be editable on completion. How defaults work in our architecture was not well considered on creation. Hopefully we can refactor user and farm level defaults in the future.